sklearn/cluster/optics_.py

```diff
@@ -0,0 +1,770 @@
+        required to store the tree. The optimal value depends on the
+        nature of the problem.
+
+    p : integer, optional (default=2)
+        Parameter for the Minkowski metric from
+        :ref:`sklearn.metrics.pairwise.pairwise_distances`. When p = 1, this is
+        equivalent to using manhattan_distance (l1), and euclidean_distance
+        (l2) for p = 2. For arbitrary p, minkowski_distance (l_p) is used.
+
+    metric_params : dict, optional (default=None)
+        Additional keyword arguments for the metric function.
+
+    n_jobs : int, optional (default=1)
+        The number of parallel jobs to run for neighbors search.
+        If ``-1``, then the number of jobs is set to the number of CPU cores.
+        Affects only :meth:`kneighbors` and :meth:`kneighbors_graph` methods.
+
+    maxima_ratio : float, optional
+        The maximum ratio we allow of average height of clusters on the
+        right and left to the local maxima in question. The higher the
+        ratio, the more generous the algorithm is to preserving local
+        minima, and the more cuts the resulting tree will have.
+
+    rejection_ratio : float, optional
+        Adjusts the fitness of the clustering. When the maxima_ratio is
+        exceeded, determine which of the clusters to the left and right to
+        reject based on rejection_ratio. Higher values will result in points
+        being more readily classified as noise; conversely, lower values will
+        result in more points being classified.
+
+    similarity_threshold : float, optional
+        Used to check if nodes can be moved up one level, that is, if the
+        new cluster created is too "similar" to its parent, given the
+        similarity threshold. Similarity can be determined by 1) the size
+        of the new cluster relative to the size of the parent node or
+        2) the average of the reachability values of the new cluster
+        relative to the average of the reachability values of the parent
+        node. A lower value for the similarity threshold means less levels
+        in the tree.
+
+    significant_min : float, optional
+        Sets a lower threshold on how small a significant maxima can be
+
+    min_cluster_size_ratio : float, optional
+        Minimum percentage of dataset expected for cluster membership.
+
+    min_maxima_ratio : float, optional
+        Used to determine neighborhood size for minimum cluster membership.
+
+    Attributes
+    ----------
+    `core_sample_indices_` : array, shape (n_core_samples,)
+        Indices of core samples.
+
+    `labels_` : array, shape (n_samples,)
+        Cluster labels for each point in the dataset given to fit().
+        Noisy samples are given the label -1.
+
+    `reachability_` : array, shape (n_samples,)
+        Reachability distances per sample.
+
+    `ordering_` : array, shape (n_samples,)
+        The cluster ordered list of sample indices
+
+    `core_dists_` : array, shape (n_samples,)
```

Comment:
perhaps rename to `core_distance_` or `core_distances_` as our users also deal a lot with "distributions"...

Reply:
changed.